docs: document EKS Auto discovery self-bootstrap (#42107)
gravitational/teleport: docs/pages/auto-discovery/kubernetes.mdx

Reviewer: We should remove the `layout: tocless-doc` frontmatter item, since with the wider layout and no table of contents, this page is kind of difficult to read.

       # Optional section: Defaults to "*":"*"
       tags:
         "env": "prod"
       # AWS role to assume when discovering resources in the AWS Account.
       # This value is an optional AWS role ARN to assume when polling EKS clusters
       assume_role_arn: arn:aws:iam::123456789012:role/iam-discovery-role
       # External Id is an optional value that should be set when accessing 
       # your AWS account from a 3rd party access (delegate access).

>        external_id: 123456789

Reviewer 2: ```suggestion
       external_id: "example-external-id"
```

Match to the docs/reference/config/ for Kubenertes service

    resources:
    - labels:
        "*": "*" # can be configured to limit the clusters to watched by this service.
      aws:
       # AWS role to assume when accessing EKS clusters in the AWS Account.
       # This value is an optional AWS role ARN to assume when forwarding requests
       # to EKS clusters.
       assume_role_arn: arn:aws:iam::123456789012:role/iam-discovery-role
       # External ID is an optional value that should be set when accessing
	     # your AWS account from a third-party service (delegated access).
       external_id: "example-external-id"
```

When configuring the `kubernetes_service.resources` parameter, the Teleport
Kubernetes Service is set to monitor EKS clusters that are discovered by
the Teleport Discovery Service. The monitoring process involves a label
matching mechanism to identify and manage the EKS clusters.

> 1. **Discovery and Label Matching**:
>   - The Discovery Service identifies available EKS clusters within the AWS environment.
>   - The Teleport Kubernetes Service checks the labels of these discovered EKS clusters
>   against the labels specified in the `kubernetes_service.resources[].labels` configuration.
>   - The labels in the `kubernetes_service.resources[].labels` array serve as selectors.
> 2. **Role Selection**:
>   - The first selector in the array that matches the labels of an EKS cluster determines
>   the role that the Kubernetes Service will assume.
>   - This role is essential for the Teleport Kubernetes Service to retrieve necessary cluster details from the AWS API.
>   - If no match is found, the Kubernetes Service defaults to its own identity.
> 3. **Interaction with AWS and Kubernetes API**:
>   - Once a match is found and a role is assumed, the Teleport Kubernetes Service uses
>   this role to access the AWS API.
>   - It retrieves detailed information about the EKS cluster, such as configuration and status.
>   - Subsequently, the Teleport Kubernetes Service forwards requests to the Kubernetes
>   API, enabling interaction with the cluster.

Reviewer: ```suggestion
1. **Discovery and Label Matching:** The Discovery Service identifies available
   EKS clusters within the AWS environment. The Teleport Kubernetes Service
   checks the labels of these clusters against the labels specified in the
   `kubernetes_service.resources[].labels` configuration, which is the
   **selector array**.
1. **Role Selection:** The first selector in the array that matches the labels
   of an EKS cluster determines the role that the Kubernetes Service will
   assume. This role is essential for the Teleport Kubernetes Service to
   retrieve necessary cluster details from the AWS API. If no match is found,
   the Kubernetes Service defaults to its own identity.
1. **Interaction with AWS and Kubernetes APIs:** Once a match is found and a
   role is assumed, the Teleport Kubernetes Service uses this role to access the
   AWS API. It retrieves information about the EKS cluster, such as
   configuration and status. Subsequently, the Teleport Kubernetes Service
   forwards requests to the Kubernetes API, enabling interaction with the
   cluster.
```

This list wasn't rendering as a list in the docs engine, which I think explains the formatting issues I noticed, so I've suggested formatting it as an ordered list and removing the bullets.

    resources:
    - labels:
        "*": "*" # can be configured to limit the clusters to watched by this service.
      aws:
       # AWS role to assume when accessing EKS clusters in the AWS Account.
       # This value is an optional AWS role ARN to assume when forwarding requests
       # to EKS clusters.
       assume_role_arn: arn:aws:iam::123456789012:role/iam-discovery-role
       # External Id is an optional value that should be set when accessing
       # your AWS account from a 3rd party access (delegate access).
       external_id: "example-external-id"
```

When configuring the `kubernetes_service.resources` parameter, the Teleport
Kubernetes Service is set to monitor EKS clusters that are discovered by
the Teleport Discovery Service. The monitoring process involves a label
matching mechanism to identify and manage the EKS clusters.

> 1. **Discovery and Label Matching**:

Reviewer: This section is a little difficult for me to read. Would it be possible to use subheadings in place of the numbered list items and regular body text instead of the bulleted list items? Since these are full sentences, I don't think bullets make them more readable.

Reviewer: I'll make a suggestion!

    resources:
    - labels:
        "*": "*" # can be configured to limit the clusters to watched by this service.
      aws:
       # AWS role to assume when accessing EKS clusters in the AWS Account.
       # This value is an optional AWS role ARN to assume when forwarding requests
       # to EKS clusters.
       assume_role_arn: arn:aws:iam::123456789012:role/iam-discovery-role
       # External Id is an optional value that should be set when accessing
       # your AWS account from a 3rd party access (delegate access).

>        external_id: 123456789

Reviewer 2: ```suggestion
       external_id: "example-external-id"
```

same as above
